### PR TITLE
[Feature] Create Petition

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add Create Petition functionality

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
I see there is a merge conflict and missing screenshots @gertie-sheshe

**Review by KobbyMmo — COMMENTED** [review]
Tried creating the petition online, but didn't go through. I think we should display error messages if the petition fails.

**Comment by gertie-sheshe** [comment]
> Tried creating the petition online, but didn't go through. I think we should display error messages if the petition fails.

Hey @esirK Is it possible to send back a different HTTP Status code when the data is invalid for the backend? Currently you are sending it as 200 ok and the error message looking like the data

**Comment by esirK** [comment]
> > Tried creating the petition online, but didn't go through. I think we should display error messages if the petition fails.
> 
> Hey @esirK Is it possible to send back a different HTTP Status code when the data is invalid for the backend? Currently you are sending it as 200 ok and the error message looking like the data

Hey @gertie-sheshe sure I can add that. Could you share an example that returns `200` for invalid data? As an example the following returns `400` in case some fields are missing.
Screenshot 2022-03-14 at 09 25 07

File: src/pages/act-now.js [hunk 1]
```diff
@@ -38,7 +40,7 @@ function ActNow({ ...props }) {
 
   // If no session exists, show default landing page
   if (!session) {
-    return <ActNowPage {...props} />;
+    return <ActNowPage session={session} {...props} />;
```
Comment (KhadijaMahanga) [inline src/pages/act-now.js:43]: why are we passing a session while it doesn't exist?

File: src/components/FormDialog/index.js [hunk 2]
```diff
@@ -16,11 +16,52 @@ import useStyles from "./useStyles";
...
+        Authorization: `Bearer ${session.accessToken}`,
+      },
+      body: JSON.stringify(values),
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw new Error("Network response was not ok.");
+      })
```
Comment (KhadijaMahanga) [inline src/components/FormDialog/index.js:60]: the lib function to create does something similar (i.e passing bearer using session from api request)

**Comment by kilemensi** [comment]
@KobbyMmo / @gertie-sheshe 

----

1. Tried to create petitions twice but I don't see any error and I don't see my petitions when I go to the actNOW page or even in `/analysis/petitions` page.
2. Image upload says you can drag and drop image but drag and drop doesn't work for me (FF on Ubuntu).

**Comment by gertie-sheshe** [comment]
> @gertie-sheshe this looks okay. Though I can't successfully create a petition. I get `Something went wrong. Please try again`. And upon investigating, its a 401/unauthorised error.
> 
> ```
> {
>     "detail": "Given token not valid for any token type",
>     "code": "token_not_valid",
>     "messages": [
>         {
>             "token_class": "AccessToken",
>             "token_type": "access",
>             "message": "Token is invalid or expired"
>         }
>     ]
> }
> ```
> 
> Where you able to create a petition successfully?

Oooh ooh okay. Yeah I am able to create one, were you already registered? If so, clear the session from the browser then try again - I think the refresh has a problem - I noticed that @KhadijaMahanga

**Review by KhadijaMahanga — APPROVED** [review]
🚀 🚀

**Pull request merged**